source/Calamari.Aws/DeployAwsCloudFormationCommand.cs

```diff
@@ -0,0 +1,173 @@
+
+                    SetOutputVariables(await cloudFormationService.GetOutputVariablesByStackArn(result.Stack));
+                }
+            }
+            else
+            {
+                var isRollbackDisabled = variables.GetFlag(SpecialVariableNames.Action.DisableRollBack, true);
+
+                var stackId = await cloudFormationService.Deploy(cloudFormationTemplate, stackArn, roleArn, iamCapabilities, isRollbackDisabled, waitForCompletion);
+                // Take the stackArn ID returned by the create or update events, and save it as an output variable
+                SetOutputVariable("StackId", stackId);
+                SetOutputVariables(await cloudFormationService.GetOutputVariablesByStackArn(stackArn));
+            }
+        }
+
+        //TODO: Refactor ITemplateResolver in Calamari.Common to make it a generic ITemplateResolver<TypeTemplate> so it returns the template directly 
+        CloudFormationTemplate GetCloudFormationTemplate()
+        {
+            var isTemplateFilesInPackage = variables.Get(SpecialVariableNames.Aws.CloudFormation.TemplateSource).Equals(SpecialVariableValues.CloudFormation.TemplateSource.Package, StringComparison.InvariantCultureIgnoreCase);
+
+            var template = variables.Get(SpecialVariableNames.Aws.CloudFormation.Template);
+
+            if (isTemplateFilesInPackage)
+            {
+                var templateResolver = new TemplateResolver(fileSystem);
+
+                var resolvedTemplate = templateResolver.Resolve(template, true, variables);
+
+                var templateParameterFile = variables.Get(SpecialVariableNames.Aws.CloudFormation.TemplateParametersRaw);
+                var resolvedParameters = templateResolver.MaybeResolve(templateParameterFile, true, variables);
+                if (templateParameterFile != null && !resolvedParameters.Some())
+                    throw new CommandException("Could not find template parameters file: " + templateParameterFile);
+
+                var parameters = CloudFormationParametersFile.Create(resolvedParameters, fileSystem, variables);
+
+                return CloudFormationTemplate.Create(resolvedTemplate, parameters, fileSystem, variables);
+            }
+            else
+            {
+                var parameters = new CloudFormationParametersFile(
+                    () => Maybe<string>.Some(variables.Get(SpecialVariableNames.Aws.CloudFormation.TemplateParameters) ?? string.Empty),
+                    JsonConvert.DeserializeObject<List<Parameter>>);
+
+                return new CloudFormationTemplate(() => template, parameters, JsonConvert.DeserializeObject<List<StackFormationNamedOutput>>);
+            }
+        }
+
+        bool IsImmediateChangeSetExecution()
+        {
+            return !bool.TrueString.Equals(variables.Get(SpecialVariableNames.Aws.CloudFormation.ChangeSets.Defer), StringComparison.InvariantCultureIgnoreCase);
```

Comment:
We should use `Ordinal` over `Invariant` because of https://github.com/OctopusDeploy/Calamari/pull/485